src/lib/StyleSelectControl.svelte

```diff
@@ -0,0 +1,239 @@
+<!-- A component to render a style selection prompt, including a thumbnail
```

Comment:
Can we move the JS script in this file to a `.js` file instead of a `.svelte`file?

It will be much easier to integrate into other frontend frameworks (e.g. React) if the MapLibre lifecycle management were done in pure JS.

Reply:
Sounds like a good idea!

Do you mean to move this logic perhaps into `styles.ts` & export the functions for use in this component:

https://github.com/dimfeld/svelte-maplibre/blob/c39d6422cf2b68f5c198c579fde3bbfecac140a3/src/lib/StyleSelectControl.svelte#L46C4-L123C4

Then the component will basically just accept props and pass data to these functions - is that right?